server/storage/wal/wal_test.go

```diff
@@ -96,6 +96,73 @@ func TestNew(t *testing.T) {
 	}
 }
 
+func TestCreateNewWALFile(t *testing.T) {
+	tests := []struct {
+		name     string
+		fileType interface{}
+		forceNew bool
+	}{
+		{
+			name:     "creating standard file should succeed and not truncate file",
+			fileType: &os.File{},
+			forceNew: false,
+		},
+		{
+			name:     "creating locked file should succeed and not truncate file",
+			fileType: &fileutil.LockedFile{},
+			forceNew: false,
+		},
+		{
+			name:     "creating standard file with forceNew should truncate file",
+			fileType: &os.File{},
+			forceNew: true,
+		},
+		{
+			name:     "creating locked file with forceNew should truncate file",
+			fileType: &fileutil.LockedFile{},
+			forceNew: true,
+		},
+	}
+
+	for i, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			p := filepath.Join(t.TempDir(), walName(0, uint64(i)))
+
+			// create initial file with some data to verify truncate behavior
+			err := os.WriteFile(p, []byte("test data"), fileutil.PrivateFileMode)
+			require.NoError(t, err)
+
+			var f interface{}
+			switch tt.fileType.(type) {
+			case *os.File:
+				f, err = createNewWALFile[*os.File](p, tt.forceNew)
+				require.IsType(t, &os.File{}, f)
+			case *fileutil.LockedFile:
+				f, err = createNewWALFile[*fileutil.LockedFile](p, tt.forceNew)
+				require.IsType(t, &fileutil.LockedFile{}, f)
```

Comment:
```suggestion
			case *os.File:
				f, err = createNewWALFile[*os.File](p, tt.forceNew)
				require.IsType(t, &os.File{}, f)
			case *fileutil.LockedFile:
				f, err = createNewWALFile[*fileutil.LockedFile](p, tt.forceNew)
				require.IsType(t, &fileutil.LockedFile{}, f)
			default:	
				panic("unknown file type")					
```

Reply:
Added a default case to handle unknown file types. Good catch on handling that scenario